### PR TITLE
feat: add support for ACCOUNT_TYPE=NEXUS

### PR DESCRIPTION
**Pull request by highskore**
### Summary
Adds support for `ACCOUNT_TYPE=NEXUS` when running tests.
### Changes
1. **Imported Nexus**:
   - Added [Nexus](https://github.com/bcnmy/nexus/tree/main) module to package.
2. **Added Precompiled Contracts**:
   - Added necessary precompiled contracts from Nexus
   - Manually used the `--via-ir` flag to compile these contracts and generate bytecode, currently the `modulekit` repository does not currently compile with `--via-ir`.
3. **Added `NexusFactory` based on other factory contracts**
   - Deploys precompiles and whitelists factory, implements the `IAccountFactory` interface
### Testing
  - **Run Tests**:
    ```sh
    ACCOUNT_TYPE=NEXUS forge test -vvvv

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,7 +41,8 @@
     "excessively-safe-call": "github:nomad-xyz/ExcessivelySafeCall",
     "forge-std": "github:foundry-rs/forge-std",
     "solady": "github:vectorized/solady",
-    "solarray": "github:sablier-labs/solarray"
+    "solarray": "github:sablier-labs/solarray",
+    "@bcnmy/nexus": "github:bcnmy/nexus"
```
Comment (kopy-kat) [inline package.json:45]: we can remove this now right

File: remappings.txt [hunk 2]
```diff
@@ -14,8 +13,11 @@ account-abstraction-v0.6/=node_modules/@ERC4337/account-abstraction-v0.6/contrac
...
 solarray/=node_modules/solarray/src/
 @prb/math/=node_modules/@prb/math/src/
 kernel/=node_modules/@zerodev/kernel/src/
-ExcessivelySafeCall/=node_modules/excessively-safe-call/src/
\ No newline at end of file
+ExcessivelySafeCall/=node_modules/excessively-safe-call/src/
+solady/=node_modules/solady/src
+sentinellist/=node_modules/@rhinestone/sentinellist/src/
+@nexus=node_modules/@bcnmy/nexus/contracts/
+node_modules/@bcnmy/nexus/contracts/:account-abstraction=node_modules/@ERC4337/account-abstraction/
```
Comment (kopy-kat) [inline remappings.txt:23]: same here

File: package.json (hunk 1 above)
Comment (kopy-kat) [reply package.json:45]: I guess we can also keep it - the interfaces below wouldnt really need to be used so we could save on an extra dependency but up to you

File: src/accounts/nexus/NexusFactory.sol [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    IBiconomyMetaFactory internal biconomyFactory;
+    NexusPrecompiles internal precompiles;
+
+    function init() public override {
+        precompiles = new NexusPrecompiles();
+        // Deploy precompiled contracts
+        nexusImpl = precompiles.deployNexus(ENTRYPOINT_ADDR);
+        factory = precompiles.deployNexusAccountFactory(address(nexusImpl), address(this));
+        bootstrapDefault = precompiles.deployNexusBootstrap();
+        biconomyFactory = precompiles.deployBiconomyMetaFactory(address(this));
```
Comment (kopy-kat) [inline src/accounts/nexus/NexusFactory.sol:33]: what is the `biconomyFactory` used for?

File: src/test/helpers/NexusHelpers.sol [hunk 4]
```diff
@@ -0,0 +1,7 @@
+// SPDX-License-Identifier: MIT
```
Comment (kopy-kat) [inline src/test/helpers/NexusHelpers.sol:1]: we can just remove this since its unused

File: src/test/helpers/interfaces/IBiconomyMetaFactory.sol [hunk 5]
```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: MIT
```
Comment (kopy-kat) [inline src/test/helpers/interfaces/IBiconomyMetaFactory.sol:1]: nit: imo all of these interfaces would fit better in `src/accounts/nexus`

File: src/test/helpers/NexusHelpers.sol (hunk 4 above)
Comment (highskore) [reply src/test/helpers/NexusHelpers.sol]: removed in [0972520](https://github.com/rhinestonewtf/modulekit/pull/135/commits/0972520f18149f3cfd4e574311af6ae5a45deaac)

File: src/accounts/nexus/NexusFactory.sol (hunk 3 above)
Comment (highskore) [reply src/accounts/nexus/NexusFactory.sol]: nothing atm, removed in [0972520](https://github.com/rhinestonewtf/modulekit/pull/135/commits/0972520f18149f3cfd4e574311af6ae5a45deaac)

File: package.json (hunk 1 above)
Comment (highskore) [reply package.json]: removed in [0972520](https://github.com/rhinestonewtf/modulekit/pull/135/commits/0972520f18149f3cfd4e574311af6ae5a45deaac)

**Pull request merged**